src/app/admin-patients-list/admin-patients-list.component.ts

```diff
@@ -0,0 +1,79 @@
+import { Component, OnInit, Input, Inject } from '@angular/core';
+import { PatientResponse } from '../../../api-objects/PatientResponse';
+import { CustomerSearchService } from '../customer-search/customer-search.service';
+import { MatDialog, MAT_DIALOG_DATA } from '@angular/material';
+import { AdminPatientListModalComponent } from './admin-patient-list-modal/admin-patient-list-modal.component';
+
+@Component({
+  selector: 'admin-patients-list',
+  templateUrl: './admin-patients-list.component.html',
+  styleUrls: ['./admin-patients-list.component.css']
+})
+export class AdminPatientsListComponent implements OnInit {
+  public patients: Array<PatientResponse> = [];
+  public currentPage: number = 0;
+
+  constructor(
+    public dialog: MatDialog
+  ) { }
+
+  public ngOnInit() {
+  }
+
+  public deletePatient(patient: PatientResponse) {
+    const dialogRef = this.dialog.open(AdminPatientListModalComponent, { data: { patientid: patient.patientid } });
+    dialogRef.afterClosed().subscribe((result) => {
+      if (result === 'confirm') {
+        // delete this patient and all of its appointments
+        console.log(`Deleting ${patient.patientid}`);
+      } else {
+        console.log('Cancel operation');
+      }
+    });
+  }
+
+  public loadPatient(patient: PatientResponse) {
+    // Load version of customer search with this patient ID and ability to delete an appointment
+    console.log('load patient: ' + patient.patientid);
+  }
+
+  public loadPatients() {
+    this.currentPage = 0;
+    // TODO: Load patients from server
+    this.patients = new Array<PatientResponse>();
+    let patient: PatientResponse;
+    for (let i = 0; i < 20; i++) {
+      patient = new Object as PatientResponse;
+      patient.deceased = Math.random() >= 0.5;
+      patient.patientid = Math.floor(Math.random() * 10000);
+      patient.patientnotes = Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36) + ' ' + Math.random().toString(36);
+      this.patients.push(patient);
+    }
+    this.patients.sort((a: PatientResponse, b: PatientResponse) => {
```

Comment:
We can probably have the API output sorted data at no performance cost, since you're sorting by serial primary key. But I suppose this just becomes O(n) if we do.

Reply:
We could actually push this functionality to Angular: https://material.angular.io/components/sort/overview
I think it would be really useful if we could sort by data rows other than Patient id.